### ISSUE TITLE
jamovi versioning issues

### ISSUE THREAD

**Issue by flammable** [autopkg/n8felton-recipes#162]
Hey Nate -

Thanks for the jamovi recipes! I'm running into an issue that might have existed in earlier versions of the software. The developer hasn't been updating `CFBundleShortVersionString` with each release, so Munki thinks that version 2.3.2.0 is actually 2.3.0.0. I downloaded the "solid" release and it has the same issues - `CFBundleShortVersionString` is 2.0.0.0, but the version of the app is actually 2.2.5.0.

Here's the AutoPkg output:

```
% autopkg run -v local.munki.jamovi
Processing local.munki.jamovi...
URLTextSearcher
URLTextSearcher: Found matching text (download): /downloads/jamovi-2.3.2.0-macos.dmg
URLTextSearcher: Found matching text (match): /downloads/jamovi-2.3.2.0-macos.dmg
com.github.n8felton.shared/RemoteFilenameFinder
URLDownloader
URLDownloader: Item at URL is unchanged.
URLDownloader: Using existing /Users/adminuser/Library/AutoPkg/Cache/local.munki.jamovi/downloads/jamovi-2.3.2.0-macos.dmg
EndOfCheckPhase
CodeSignatureVerifier
CodeSignatureVerifier: Mounted disk image /Users/adminuser/Library/AutoPkg/Cache/local.munki.jamovi/downloads/jamovi-2.3.2.0-macos.dmg
CodeSignatureVerifier: Verifying code signature...
CodeSignatureVerifier: Deep verification enabled...
CodeSignatureVerifier: Strict verification not defined. Using codesign defaults...
CodeSignatureVerifier: /private/tmp/dmg.K5mXwj/jamovi.app: valid on disk
CodeSignatureVerifier: /private/tmp/dmg.K5mXwj/jamovi.app: satisfies its Designated Requirement
CodeSignatureVerifier: /private/tmp/dmg.K5mXwj/jamovi.app: explicit requirement satisfied
CodeSignatureVerifier: Signature is valid
MunkiImporter
MunkiImporter: Using repo lib: AutoPkgLib
MunkiImporter:         plugin: FileRepo
MunkiImporter:           repo: /Library/Management/munki_repo
MunkiImporter: Copied pkginfo to: /Library/Management/munki_repo/pkgsinfo/apps/jamovi-2.3.0.0.plist
MunkiImporter:            pkg to: /Library/Management/munki_repo/pkgs/apps/jamovi-2.3.2.0-macos-2.3.0.0.dmg
Receipt written to /Users/adminuser/Library/AutoPkg/Cache/local.munki.jamovi/receipts/local.munki-receipt-20220330-152533.plist

The following new items were imported into Munki:
    Name    Version  Catalogs  Pkginfo Path               Pkg Repo Path                          Icon Repo Path
    ----    -------  --------  ------------               -------------                          --------------
    jamovi  2.3.0.0  testing   apps/jamovi-2.3.0.0.plist  apps/jamovi-2.3.2.0-macos-2.3.0.0.dmg
```

If you also use this recipe, is this something you can adjust for in AutoPkg? I'll try reaching out to the developer, too. For now, I manually imported the latest "solid" version and made an installs array that checks the md5 hash of `/Applications/jamovi.app/Contents/MacOS/jamovi`.

Thanks!

Mike

**Comment by n8felton** [comment]
This appears to be fixed with 2.3.3.0. Hopefully it stays fixed.

```
{'Output': {'munki_importer_summary_result': {'data': {'catalogs': 'testing',
                                                       'icon_repo_path': '',
                                                       'name': 'jamovi',
                                                       'pkg_repo_path': 'apps/jamovi/jamovi-2.3.3.0-macos-2.3.3.0.dmg',
                                                       'pkginfo_path': 'apps/jamovi/jamovi-2.3.3.0.plist',
                                                       'version': '2.3.3.0'},
                                              'report_fields': ['name',
                                                                'version',
                                                                'catalogs',
                                                                'pkginfo_path',
                                                                'pkg_repo_path',
                                                                'icon_repo_path'],
                                              'summary_text': 'The following '
                                                              'new items were '
                                                              'imported into '
                                                              'Munki:'},
            'munki_info': {'_metadata': {'created_by': 'Nathan.Felton',
                                         'creation_date': datetime.datetime(2022, 4, 20, 19, 54, 39),
                                         'munki_version': '5.6.2.4398',
                                         'os_version': '12.3'},
                           'autoremove': False,
                           'catalogs': ['testing'],
                           'category': 'Math & Science',
                           'description': 'Free and open statistical software '
                                          'to bridge the gap between '
                                          'researcher and statistician.',
                           'developer': 'The jamovi project',
                           'display_name': 'jamovi',
                           'installer_item_hash': '541e50146d7e22c308e842296e9b324e0cc1e7e9dacf73b7363e407e502fdf7f',
                           'installer_item_location': 'apps/jamovi/jamovi-2.3.3.0-macos-2.3.3.0.dmg',
                           'installer_item_size': 384863,
                           'installer_type': 'copy_from_dmg',
                           'installs': [{'CFBundleIdentifier': 'org.jamovi.jamovi',
                                         'CFBundleName': 'jamovi',
                                         'CFBundleShortVersionString': '2.3.3.0',
                                         'minosversion': '10.13.0',
                                         'path': '/Applications/jamovi.app',
                                         'type': 'application',
                                         'version_comparison_key': 'CFBundleShortVersionString'}],
                           'items_to_copy': [{'destination_path': '/Applications',
                                              'source_item': 'jamovi.app'}],
                           'minimum_os_version': '10.13.0',
                           'name': 'jamovi',
                           'unattended_install': True,
                           'uninstall_method': 'remove_copied_items',
                           'uninstallable': True,
                           'version': '2.3.3.0'},
            'munki_repo_changed': True,
            'pkg_repo_path': '/Users/Shared/munki/pkgs/apps/jamovi/jamovi-2.3.3.0-macos-2.3.3.0.dmg',
            'pkginfo_repo_path': '/Users/Shared/munki/pkgsinfo/apps/jamovi/jamovi-2.3.3.0.plist'}}
Receipt written to /usr/local/autopkg/cache/com.github.n8felton.munki.jamovi/receipts/jamovi-receipt-20220420-155440.plist

The following new items were downloaded:
    Download Path                                                                                 
    -------------                                                                                 
    /usr/local/autopkg/cache/com.github.n8felton.munki.jamovi/downloads/jamovi-2.3.3.0-macos.dmg  

The following new items were imported into Munki:
    Name    Version  Catalogs  Pkginfo Path                      Pkg Repo Path                                 Icon Repo Path  
    ----    -------  --------  ------------                      -------------                                 --------------  
    jamovi  2.3.3.0  testing   apps/jamovi/jamovi-2.3.3.0.plist  apps/jamovi/jamovi-2.3.3.0-macos-2.3.3.0.dmg                  
```

**Comment by flammable** [comment]
Just getting to look at this again. I'm glad the developer was able to address that! Everything looks great here, thanks!
